test/test_autograd.py

```diff
@@ -2720,6 +2720,21 @@ def f(x, y):
 
         self.assertTrue('my_func' in str(p))
 
+    def test_record_function_multithreaded(self):
```

Comment:
In what sense is this test multithreaded?
Or is it just checking that enter and exit outside the scope of a given profile works?

Reply:
This is more testing python-style GIL multithreading than true multi-parallelism such as in C++. There's still some thread safety issues with enabling & disabling the profiler callbacks that we might need to consider when we start profiling TorchScript